spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTransformConverter.java

```diff
@@ -0,0 +1,243 @@
+  }
+
+  private static Distribution toGravitinoDistribution(BucketTransform bucketTransform) {
+    int bucketNum = (Integer) bucketTransform.numBuckets().value();
+    Expression[] expressions =
+        JavaConverters.seqAsJavaList(bucketTransform.columns()).stream()
+            .map(sparkReference -> NamedReference.field(sparkReference.fieldNames()))
+            .toArray(Expression[]::new);
+    return Distributions.hash(bucketNum, expressions);
+  }
+
+  // Spark datasourceV2 doesn't support specify sort order direction, use ASCENDING as default.
+  private static Pair<Distribution, SortOrder[]> toGravitinoDistributionAndSortOrders(
+      SortedBucketTransform sortedBucketTransform) {
+    int bucketNum = (Integer) sortedBucketTransform.numBuckets().value();
+    Expression[] bucketColumns =
+        transToGravitinoNamedReference(
+            JavaConverters.seqAsJavaList(sortedBucketTransform.columns()));
+
+    Expression[] sortColumns =
+        transToGravitinoNamedReference(
+            JavaConverters.seqAsJavaList(sortedBucketTransform.sortedColumns()));
+    SortOrder[] sortOrders =
+        Arrays.stream(sortColumns)
+            .map(
+                sortColumn ->
+                    SortOrders.of(sortColumn, ConnectorConstants.SPARK_DEFAULT_SORT_DIRECTION))
+            .toArray(SortOrder[]::new);
+
+    return Pair.of(Distributions.hash(bucketNum, bucketColumns), sortOrders);
+  }
+
+  private static org.apache.spark.sql.connector.expressions.Transform toSparkBucketTransform(
+      Distribution distribution, SortOrder[] sortOrders) {
+    if (distribution == null) {
+      return null;
+    }
+
+    switch (distribution.strategy()) {
+      case NONE:
+        return null;
+      case HASH:
+        int bucketNum = distribution.number();
+        String[] bucketFields =
+            Arrays.stream(distribution.expressions())
+                .map(
+                    expression ->
+                        getFieldNameFromGravitinoNamedReference((NamedReference) expression))
+                .toArray(String[]::new);
+        if (sortOrders == null || sortOrders.length == 0) {
+          return Expressions.bucket(bucketNum, bucketFields);
+        } else {
+          String[] sortOrderFields =
+              Arrays.stream(sortOrders)
+                  .map(
+                      sortOrder ->
+                          getFieldNameFromGravitinoNamedReference(
+                              (NamedReference) sortOrder.expression()))
+                  .toArray(String[]::new);
+          return createSortBucketTransform(bucketNum, bucketFields, sortOrderFields);
+        }
+        // Spark doesn't support EVEN or RANGE distribution
+      default:
+        throw new NotSupportedException(
+            "Doesn't support distribution strategy: " + distribution.strategy());
+    }
+  }
+
+  private static Expression[] transToGravitinoNamedReference(
```

Comment:
Maybe we can rename to "toGravitinoNamedReference"?

Reply:
done